src/kubernetes-ingress-controller/references/custom-resources.md

```diff
@@ -1,458 +1,429 @@
-
-```yaml
-apiVersion: configuration.konghq.com/v1beta1
-kind: UDPIngress
-metadata:
-  name: <object name>
-  namespace: <object namespace>
-  annotations:
-    kubernetes.io/ingress.class: <controller ingress class, "kong" by default>
-spec:
-  rules:
-  - port: <port on which to expose this service, required>
-    backend:
-      serviceName: <name of the kubernetes service, required>
-      servicePort: <port number to forward on the service, required>
-```
-
-For each rule provided in the spec the Kong proxy environment must be updated to
-listen to UDP on that port as well.
-
-[udp]:https://datatracker.ietf.org/doc/html/rfc768
-
-## KongConsumer
-
-This custom resource configures a consumer in Kong:
-
-The following snippet shows the field available in the resource:
-
-```yaml
-apiVersion: configuration.konghq.com/v1
-kind: KongConsumer
-metadata:
-  name: <object name>
-  namespace: <object namespace>
-  annotations:
-    kubernetes.io/ingress.class: <controller ingress class, "kong" by default>
-username: <user name>
-custom_id: <custom ID>
-```
-
-An example:
-
-```yaml
-apiVersion: configuration.konghq.com/v1
-kind: KongConsumer
-metadata:
-  name: consumer-team-x
-  annotations:
-    kubernetes.io/ingress.class: kong
-username: team-X
-credentials:
-  - secretRef1
-  - secretRef2
-```
-
-When this resource is created, a corresponding consumer entity will be
-created in Kong.
-
-Consumers' `username` and `custom_id` values must be unique across the Kong
-cluster. While KongConsumers exist in a specific Kubernetes namespace,
-KongConsumers from all namespaces are combined into a single Kong
-configuration, and no KongConsumers with the same `kubernetes.io/ingress.class`
-may share the same `username` or `custom_id` value.
-
-For help configuring credentials for the `KongConsumer` Please refer to the [using the Kong Consumer and Credential resource](/kubernetes-ingress-controller/{{page.kong_version}}/guides/using-consumer-credential-resource) guide.
-
-[k8s-crd]: https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/
-[kong-consumer]: /gateway/latest/admin-api/#consumer-object
-[kong-plugin]: /gateway/latest/admin-api/#plugin-object
-[kong-upstream]: /gateway/latest/admin-api/#upstream-object
-[kong-service]: /gateway/latest/admin-api/#service-object
-[kong-route]: /gateway/latest/admin-api/#route-object
+[This document is generated by KIC's 'generate.docs' make target, DO NOT EDIT]: #
+
+## Packages
+- [configuration.konghq.com/v1](#configurationkonghqcomv1)
+- [configuration.konghq.com/v1alpha1](#configurationkonghqcomv1alpha1)
+- [configuration.konghq.com/v1beta1](#configurationkonghqcomv1beta1)
+
+
+## configuration.konghq.com/v1
+
+Package v1 contains API Schema definitions for the konghq.com v1 API group.
+
+- [KongClusterPlugin](#kongclusterplugin)
+- [KongConsumer](#kongconsumer)
+- [KongIngress](#kongingress)
+- [KongPlugin](#kongplugin)
+
+
+
+### ConfigSource
```

Comment:
Could we put the internal structures in the lower place of the page?

Reply:
PTAL at https://github.com/Kong/kubernetes-ingress-controller/pull/3211